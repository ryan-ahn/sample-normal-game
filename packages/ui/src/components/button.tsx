import { Button, ButtonProps, cn, tv } from "@nextui-org/react";

export type ButtonVariant = "solid" | "bordered";
export type ButtonRounded = "none" | "tn" | "sm" | "md" | "lg" | "full";
export type ButtonSize = "sm" | "md" | "lg";
export type ButtonColor = "default" | "primary";

const button = tv({
  base: "rounded-none font-medium",
  variants: {
    variant: {
      solid: "",
      bordered: "border-1 bg-background",
    },
    rounded: {
      none: "",
      tn: "rounded-tn",
      sm: "rounded-sm",
      md: "rounded-md",
      lg: "rounded-lg",
      full: "rounded-full",
    },
    size: {
      sm: "text-sm",
      md: "text-sm font-bold",
      lg: "text-md h-14 px-4 py-3 font-bold",
    },
    color: {
      default: "bg-background",
      primary: "bg-primary",
    },
  },
  compoundVariants: [
    {
      variant: ["solid"],
      color: ["default"],
      className: "bg-foreground text-background",
    },
    {
      variant: ["bordered"],
      color: ["default"],
      className: "border-default-500 text-foreground",
    },
    {
      variant: ["bordered"],
      color: ["primary"],
      className: "border-primary-300 text-primary",
    },
  ],
  defaultVariants: {
    variant: "solid",
    rounded: "none",
    size: "md",
    color: "default",
  },
});

export const ButtonV1 = ({
  className,
  variant,
  rounded,
  size,
  color,
  as,
  to,
  children,
  ...rest
}: {
  className?: string;
  variant?: ButtonVariant;
  rounded?: ButtonRounded;
  size?: ButtonSize;
  color?: ButtonColor;
  as?: React.RefAttributes<HTMLAnchorElement>;
  to?: string;
  children: React.ReactNode;
} & ButtonProps) => {
  return (
    <Button
      as={as}
      to={to}
      className={cn(button({ variant, rounded, size, color }), className)}
      {...rest}
    >
      {children}
    </Button>
  );
};
