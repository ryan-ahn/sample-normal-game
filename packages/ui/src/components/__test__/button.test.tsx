import { render, screen } from "@testing-library/react";

import { ButtonV1 } from "../button";

describe("ButtonV1", () => {
  it("default", () => {
    render(<ButtonV1>Call to action copy</ButtonV1>);

    const button = screen.getByText("Call to action copy");
    expect(button).toMatchSnapshot();
  });
});
