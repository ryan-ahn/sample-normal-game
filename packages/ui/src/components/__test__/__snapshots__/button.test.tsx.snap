// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonV1 > default 1`] = `
<button
  class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover rounded-none text-sm font-bold bg-foreground text-background"
  type="button"
>
  Call to action copy
</button>
`;
